### PR TITLE
Added catalogue page

### PR DESCRIPTION
**Pull request by benjimarshall**
bookish_catalogue1

### CODE REVIEW

File: Bookish.DataAccess/BookishService.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@ public interface IBookishService
         IEnumerable<Book> GetBooks();
         User? GetUser(string name);
         IEnumerable<LoanedBook> GetLoanedBooks(string username);
+        public IEnumerable<CataloguedBook> GetCatalogue();
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:14]: `public` isn't necessary here 🙂

File: Bookish.DataAccess/BookishService.cs [hunk 2]
```diff
@@ -52,5 +53,21 @@ FROM loans
...
         }
+
+        public IEnumerable<CataloguedBook> GetCatalogue()
+        {
+            var sqlString =
+                @"SELECT books.title AS Title,
+                    books.authors AS Authors,
+                    books.isbn AS Isbn,
+                    COUNT(bookcopies.id) TotalCopies,
+                    COUNT(bookcopies.id) - COUNT(loans.due) AS AvailableCopies
```
Comment (FridaTveit) [inline Bookish.DataAccess/BookishService.cs:64]: Shouldn't this be `loans.bookid`? 🙂

File: Bookish.Web/Views/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -22,6 +22,9 @@
                         <li class="nav-item">
                             <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Index">Loans</a>
                         </li>
+                        <li class="nav-item">
+                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Catalogue">Catalogue</a>
+                        </li>
```
Comment (FridaTveit) [inline Bookish.Web/Views/Shared/_Layout.cshtml:27]: Does the navbar hightlight which link is selected at the moment? 🙂

**Review by FridaTveit — COMMENTED** [review]
A few comments, otherwise looks good 🙂

File: Bookish.DataAccess/BookishService.cs (hunk 2 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs:64]: As per our chat, `loans.due` is used to count whether the book is on loan (not `null`) or not on loan (`null`), to find the number of available books, compared to the total number of books

File: Bookish.DataAccess/BookishService.cs (hunk 1 above)
Comment (benjimarshall) [reply Bookish.DataAccess/BookishService.cs]: Thanks 🙂 Changed in 88b337d

File: Bookish.Web/Views/Shared/_Layout.cshtml (hunk 3 above)
Comment (benjimarshall) [reply Bookish.Web/Views/Shared/_Layout.cshtml:27]: I've just added that in 10181b3 🙂

**Comment by benjimarshall** [comment]
### New screenshot:
bookish_catalogue2

File: Bookish.Web/Views/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -19,11 +19,11 @@
...
                     <partial name="_LoginPartial" />
                     <ul class="navbar-nav flex-grow-1">
-                        <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Index">Loans</a>
+                        <li class="nav-item @(ViewContext.RouteData.Values["Action"].ToString() == "Index" ? "active" : "")">
+                            <a class="nav-link" asp-area="" asp-controller="Home" asp-action="Index">Loans</a>
                         </li>
-                        <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Catalogue">Catalogue</a>
+                        <li class="nav-item @(ViewContext.RouteData.Values["Action"].ToString() == "Catalogue" ? "active" : "")">
```
Comment (FridaTveit) [inline Bookish.Web/Views/Shared/_Layout.cshtml:25]: Might be worth extracting this to a helper method, e.g. as an extension method on `IHtmlHelper` or `ViewContext`? 🙂

**Review by FridaTveit — COMMENTED** [review]
One small suggestion, otherwise looks good 🙂

File: Bookish.Web/Views/Shared/_Layout.cshtml (hunk 4 above)
Comment (benjimarshall) [reply Bookish.Web/Views/Shared/_Layout.cshtml]: Done in 2fa3572 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**